app/src/androidMain/kotlin/de/westnordost/streetcomplete/quests/parking_charge/AddParkingChargeForm.kt

```diff
@@ -0,0 +1,196 @@
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.unit.dp
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.databinding.ComposeViewBinding
+import de.westnordost.streetcomplete.quests.AbstractOsmQuestForm
+import de.westnordost.streetcomplete.quests.AnswerItem
+import de.westnordost.streetcomplete.ui.common.dialogs.TextInputDialog
+import de.westnordost.streetcomplete.ui.util.content
+
+class AddParkingChargeForm : AbstractOsmQuestForm<ParkingChargeAnswer>() {
+
+    override val contentLayoutResId = R.layout.compose_view
+    private val binding by contentViewBinding(ComposeViewBinding::bind)
+
+    private lateinit var amountState: MutableState<String>
+    private lateinit var timeUnitState: MutableState<TimeUnit>
+    private lateinit var showDialogState: MutableState<Boolean>
+
+    override val otherAnswers: List<AnswerItem> get() = listOf(
+        AnswerItem(R.string.quest_parking_charge_varies) {
+            showDialogState.value = true
+        }
+    )
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        binding.composeViewBase.content {
+            Surface {
+                amountState = rememberSaveable { mutableStateOf("") }
+                timeUnitState = rememberSaveable { mutableStateOf(TimeUnit.HOUR) }
+                showDialogState = rememberSaveable { mutableStateOf(false) }
+
+                Column(
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .padding(16.dp),
+                    verticalArrangement = Arrangement.spacedBy(16.dp)
+                ) {
+                    // Input amount
+                    OutlinedTextField(
+                        value = amountState.value,
+                        onValueChange = {
+                            amountState.value = it
+                            checkIsFormComplete()
+                        },
+                        label = { Text(getString(R.string.quest_parking_charge_amount_label)) },
+                        placeholder = { Text("1.50") },
+                        keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Decimal),
+                        modifier = Modifier.fillMaxWidth(),
+                        singleLine = true
+                    )
+
+                    // Choose time Unit
+                    Text(
+                        text = getString(R.string.quest_parking_charge_time_unit_label),
+                        style = MaterialTheme.typography.body1
+                    )
+
+                    Row(
+                        modifier = Modifier.fillMaxWidth(),
+                        horizontalArrangement = Arrangement.spacedBy(8.dp)
+                    ) {
```

Comment:
To select a time unit, I would very much prefer a dropdown button (aka pop-up button). I already implemented a generic composable like this, have a look in `de.westnordost.streetcomplete.ui.common.DropdownButton`. See the `MaxHeightForm` for an example how it is used.

Reply:
Also, the dropdown that just presents 30min and 15min is a bit problematic. Already now I see in taginfo a plethora of other minutes defined. Maybe we'd really need another input field in which the user can define x day(s), x hour(s) or x minute(s).

(This **may** again make the input form too wide to conveniently fit into one line)